packages/core-components/src/components/AlertDisplay/AlertDisplay.test.tsx

```diff
@@ -92,4 +93,98 @@ describe('<AlertDisplay />', () => {
       expect(queryByText('(2 older messages)')).toBeInTheDocument();
     });
   });
+
+  describe('with transient message', () => {
+    const apis = [
+      [
+        alertApiRef,
+        {
+          post() {},
+          alert$() {
+            return Observable.of<AlertMessage>({
+              message: 'transient message one',
+              display: 'transient',
+            });
+          },
+        },
+      ] as const,
+    ] as const;
+
+    it('renders message and then removes it', async () => {
+      jest.useFakeTimers();
+      const { queryByText } = await renderInTestApp(
+        <TestApiProvider apis={apis}>
+          <AlertDisplay />
+        </TestApiProvider>,
+      );
+
+      expect(queryByText('transient message one')).toBeInTheDocument();
+      act(() => {
+        jest.advanceTimersByTime(5005);
+      });
+      expect(queryByText('transient message one')).not.toBeInTheDocument();
+      jest.useRealTimers();
+    });
+
+    it('respects transientTimeoutMs prop', async () => {
+      jest.useFakeTimers();
+      const { queryByText } = await renderInTestApp(
+        <TestApiProvider apis={apis}>
+          <AlertDisplay transientTimeoutMs={2500} />
+        </TestApiProvider>,
+      );
+
+      expect(queryByText('transient message one')).toBeInTheDocument();
+      act(() => {
+        jest.advanceTimersByTime(2505);
+      });
+      expect(queryByText('transient message one')).not.toBeInTheDocument();
+      jest.useRealTimers();
+    });
+  });
+
+  // What I'd like to do here is be able to trigger a few messages with a bit of time between each
+  // Then be able to check that they close one after the other, just not sure how to set that up?
+  describe('with multiple transient messages', () => {
```

Comment:
@Rugvip, so as the comment says I want to test multiple transient messages coming in but with some time between them so I can test to make sure they each close in the right order. Once I have this working then I'd like to add another test with mixed messages

Reply:
You'll probably need to use a bit more of a lower level method for creating the observable. I think something like this should allow you to create a reliable tests as long as you mock timers: 

```ts
const alert$ = new Observable(subscriber => {
  subscriber.next([alert]);
  setTimeout(() => {
    subscriber.next([alert, alert]);
  }, 1000)
  setTimeout(() => {
    subscriber.next([alert, alert, alert]);
  }, 2000)
})
```

Reply:
Thanks this was very helpful. I've refactored the one test I had started and added another with mixed display types.